extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/media/GoogleMediaImporter.java

```diff
@@ -143,6 +203,9 @@ public GoogleMediaImporter(
     this.connectionProvider = connectionProvider;
     this.monitor = monitor;
     this.writesPerSecond = writesPerSecond;
+    this.appCredentials = appCredentials;
+    this.retryingIdempotentExecutor = retryingIdempotentExecutor;
+    this.enableRetrying = enableRetrying;
   }
 
   @Override
```

Comment:
in this importItem call, I think the quick change you need to do to actually enable and use the executor that you passed into the constructor instead of the executor that's passed into the parameters list. 

to achieve this, the first thing in the function should be something like 

`if (enableRetrying && retryingIdempotentexecutor != null) idempotentImportExecutor = retryingIdempotentexecutor`

That would then override the value of the one that is passed in and the retrying one will be used instead (this is done in the same way for Flickr: https://github.com/google/data-transfer-project/blob/237fcc60e528a8b5d9d06f082faf5b76cd73d642/extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java#L129-L130

Reply:
sorry this is just a PR I'm using for testing, is it showing up in the main google-data-transfer-project repo PRs? 

I do appreciate you pointing these out, I'll implement them in the media exporter and send it as a PR on the main repo